addons/account/report/account_general_ledger.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Odoo, Open Source Management Solution
+#    Copyright (C) 2004-2014 OpenErp S.A. (<http://odoo.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, _
+
+
+class report_account_general_ledger(models.AbstractModel):
+    _name = "account.general.ledger"
+    _description = "General Ledger Report"
+
+    @api.model
+    def get_lines(self, context_id, line_id=None):
```

Comment:
the fact that get_lines() needs to be copied/pasted for each new custom reports makes me think that it should be defined at a higher level, in a class that all custom reports would inherit... and generic might inherit that part also,  don't you think?